java/core/src/main/java/co/worklytics/psoxy/rules/RulesUtils.java

```diff
@@ -32,23 +33,36 @@ public String sha(RuleSet rules) {
      */
     @SneakyThrows
     public Optional<RuleSet> getRulesFromConfig(ConfigService config) {
-        return config.getConfigPropertyAsOptional(ProxyConfigProperty.RULES)
-            .map(base64encoded -> Base64.getDecoder().decode(base64encoded))
-            .map(yamlString -> {
-                try {
-                    CsvRules rules = yamlMapper.readerFor(CsvRules.class).readValue(yamlString);
-                    Validator.validate(rules);
-                    return rules;
-                } catch (IOException e) {
-                    try {
-                        Rules2 rules = yamlMapper.readerFor(Rules2.class).readValue(yamlString);
-                        Validator.validate(rules);
-                        return rules;
-                    } catch (IOException ex) {
-                        throw new IllegalStateException("Invalid rules configured", e);
-                    }
-                }
-            });
+        Optional<String> configuredRules = config.getConfigPropertyAsOptional(ProxyConfigProperty.RULES);
+
+        if (configuredRules.isPresent()) {
+            String yamlEncodedRules;
+            try {
+                yamlEncodedRules = new String(Base64.getDecoder().decode(configuredRules.get()));
+            } catch (IllegalArgumentException e) {
+                yamlEncodedRules = configuredRules.get();
+            }
+            return Optional.of(parse(yamlEncodedRules));
+        } else {
+            return Optional.empty();
+        }
+    }
+
+    @VisibleForTesting
+    RuleSet parse(String yamlString) {
```

Comment:
```suggestion
    RuleSet parse(@NotNullString yamlString) {
```

otherwise `reader` will complain with its internal null check